platform/linux-generic/odp_packet.c

```diff
@@ -2052,6 +2083,18 @@ static inline uint8_t parse_ipv4(packet_parser_t *prs, const uint8_t **parseptr,
 	*offset   += ihl * 4;
 	*parseptr += ihl * 4;
 
+	if (chksums.chksum.udp || chksums.chksum.tcp) {
+		l4_part_sum = _odp_chksum_ones_comp16_32(
+				(const uint16_t *)&ipv4->src_addr,
+				2 * _ODP_IPV4ADDR_LEN, false);
+#if ODP_BYTE_ORDER == ODP_BIG_ENDIAN
+		l4_part_sum += ipv4->proto;
+#else
+		l4_part_sum += ((uint16_t)ipv4->proto) << 8;
+#endif
```

Comment:
`ipv4->proto` is a `uint8_t` field here, so why is this needed? These are the same independent of endianness.

Reply:
Yes, but the adding word is not (proto is the lat byte in pseudo header).

Reply:
The protocol field in the pseudo-header is simply the 8-bit IP protocol field considered as a 16-bit value. It's also in network byte order so these two have the same bit value.

Reply:
Yes. However depending on the endianness of the host, it should be converted to a different host u16 value. It is a pair `{0, proto}`. On BE it becomes just `proto` u16, on LE it is equal to `proto << 8`.